File: Jessica/101.Symmetric Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool isSymmetric(TreeNode* root) {
        return Recursion(root,root);
        
    }
    bool Recursion(TreeNode* n1,TreeNode* n2){
        if(n1==NULL && n2==NULL)
            return true;
        else if (n1==NULL || n2==NULL)
            return false;
        else{
            if(n1->val != n2->val)
                return false;
            else{
                return (Recursion(n1->left, n2->right) && (Recursion(n1->right, n2->left)));
            }
        }
    }
};

File: Jessica/101.Symmetric Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        if root is None:
            return True
        return self.Recursion(root.left,root.right)
    def Recursion(self,n1,n2):
        if n1 is None and n2 is None:
            return True
        elif n1 is None or n2 is None:
            return False
        elif n1.val != n2.val:
            return False
        else:
            return self.Recursion(n1.left,n2.right) and self.Recursion(n1.right,n2.left)
        
       
